designer/report-designer-extension-pentaho/src/main/java/org/pentaho/reporting/designer/extensions/pentaho/repository/util/PublishUtil.java

```diff
@@ -144,17 +140,18 @@ public static void launchReportOnServer( final String baseUrl, final String path
     }
   }
 
-  public static int publish( final byte[] data, final String path, final AuthenticationData loginData )
-    throws IOException {
+  public static int publish( final byte[] data, final String path, final AuthenticationData loginData, final Optional<Properties> fileProperties ) throws IOException {
     int responseCode = HTTP_RESPONSE_FAIL;
     final String versionText = loginData.getOption( SERVER_VERSION );
     final int version = ParserUtil.parseInt( versionText, SERVER_VERSION_SUGAR );
 
     if ( SERVER_VERSION_SUGAR == version ) {
-      PublishRestUtil publishRestUtil =
-          new PublishRestUtil( loginData.getUrl(), loginData.getUsername(), loginData.getPassword() );
-      responseCode = publishRestUtil.publishFile( path, data, true );
 
+      //TODO do not catch why we add overwrite file here but it was so, keep it as is until next review
+      fileProperties.orElseGet( () -> new Properties() ).setProperty( PublishRestUtil.OVERWRITE_FILE_KEY, Boolean.TRUE.toString() );
```

Comment:
Replace that comment with "Force overwrite flag here so that the server does not fail with an error in case the report already exists in the JCR".